src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventSearchControllerTest.java

```diff
@@ -108,14 +104,22 @@ void returnsAllFieldsCorrectly() throws Exception {
             .andReturn();
 
         var response = json.from(mvcResult.getResponse().getContentAsString());
-        assertThat(response).hasJsonPathNumberValue("[0].id");
-        assertThat(response).hasJsonPathStringValue("[0].event_ts");
-        assertThat(response).hasJsonPathNumberValue("[0].courthouse.id");
-        assertThat(response).hasJsonPathStringValue("[0].courthouse.display_name");
-        assertThat(response).hasJsonPathNumberValue("[0].courtroom.id");
-        assertThat(response).hasJsonPathStringValue("[0].courtroom.name");
-        assertThat(response).hasJsonPathBooleanValue("[0].is_event_anonymised");
-        assertThat(response).hasJsonPathBooleanValue("[0].is_case_expired");
-        assertThat(response).hasJsonPathStringValue("[0].case_expired_at");
+        assertThat(response).hasJsonPathNumberValue("$.length()", eventHearingsCount + eventsCount);
+
+        // assert the event and hearings
+        for (int i = 0; i < eventsCount; i++) {
+            for (int j = 0; j < eventHearingsCount; j++) {
+                assertThat(response).hasJsonPathNumberValue("[" + i + "].id", entity.get(i).getId());
+                assertThat(response).hasJsonPathStringValue("[" + i + "].event_ts", entity.get(i).getTimestamp().toString());
+                assertThat(response).hasJsonPathNumberValue("[" + i + "].courthouse.id", entity.get(i).getCourtroom().getCourthouse().getId());
+                assertThat(response).hasJsonPathStringValue("[" + i + "].courthouse.display_name",
+                                                            entity.get(i).getCourtroom().getCourthouse().getDisplayName());
+                assertThat(response).hasJsonPathNumberValue("[" + i + "].courtroom.id", entity.get(i).getCourtroom().getId());
+                assertThat(response).hasJsonPathStringValue("[" + i + "].courtroom.name", entity.get(i).getCourtroom().getName());
+                assertThat(response).hasJsonPathBooleanValue("[" + i + "].is_data_anonymised", entity.get(i).isDataAnonymised());
+                assertThat(response).hasJsonPathBooleanValue("[" + i + "].is_case_expired",
+                                                             entity.get(i).getHearingEntities().get(j).getCourtCase().isDataAnonymised());
+            }
+        }
     }
```

Comment:
these changes to the test code to setup multiple hearings for an event and verify through a loop are not needed once we action the other comment I've left about removal of `is_case_expired` and `case_expired_at`.
So this new test code we have introduced is not really needed